R/calculate-effect-sizes.R

```diff
@@ -353,9 +368,91 @@ batch_calc_ES <- function(dat,
                       intervals = .data[[intervals]],
                       observation_length = .data[[observation_length]],
                       confidence = confidence, 
-                      format = format,
+                      format = "long",
                       ...,
                       warn = warn)) %>%
     dplyr::ungroup()
+  
+  if (is.null(aggregate)) {
+    
+    if (format == "long") {
+      ES_ests <- ES_ests_long
+    } else {
+      ES_ests <- 
+        ES_ests_long %>% 
+        tidyr::pivot_wider(
+          names_from = ES,
+          names_glue = "{ES}_{.value}",
+          values_from = - c(!!!grouping, ES)
+        ) 
+    }
+    
+    result <- ES_ests
+    
+  } else {
+    
+    Est <- SE <- n <- CI_lower <- CI_upper <- NULL
+    
+    if (weighting == "1/V") { 

```

Comment:
Handle weights by first computing the weighting variable (depending on the `weighting` argument), then summarizing using weighted averages.